moduleframework/dockerlinter.py

```diff
@@ -161,11 +167,95 @@ def get_specific_label(self, label_name=None):
         if label_name is None:
             return []
         label_list = self.get_docker_labels()
-        return [label_name in label_list]
+        return [label_list[key] for key in label_list.keys() if label_name == key]
 
-    def check_baseruntime(self):
+    def check_from_is_first(self):
         """
-        Function returns docker labels
-        :return: label dictionary
+        Function checks if FROM directive is really first directive.
+        :return: True if FROM is first, False if FROM is not first directive
+        """
+        if self.dfp_structure[0].get('instruction') == 'FROM':
+            return True
+        else:
+            return False
+
+    def check_from_directive_is_valid(self):
+        """
+        Function checks if FROM directive contains valid format like is specified here
+        http://docs.projectatomic.io/container-best-practices/#_line_rule_section
+        Regular expression is: ^[a-z0-9.]+(\/[a-z0-9\D.]+)+$
+        Example registry:
+            registry.fedoraproject.org/f26/etcd
+            registry.fedoraproject.org/f26/flannel
+            registry.access.redhat.com/rhscl/nginx-18-rhel7
+            registry.access.redhat.com/rhel7/rhel-tools
+            registry.access.redhat.com/rhscl/postgresql-95-rhel7
+
+        :return:
+        """
+        correct_format = False
+        struct = self.dfp_structure[0]
+        if struct.get(INSTRUCT) == 'FROM':
+            p = re.compile("^[a-z0-9.]+(\/[a-z0-9\D.]+)+$")
+            if p.search(struct.get('value')) is not None:
+                correct_format = True
+        return correct_format
+
+    def check_chained_run_dnf_commands(self):
+        """
+        Function checks if Dockerfile does not contain more `RUN dnf` commands
+        in more then one row.
+        BAD examples:
+             FROM fedora
+             RUN dnf install foobar1
+             RUN dnf clean all
+        GOOD example:
+             FROM fedora
+             RUN dnf install foobar1 && dnf clean all
+        :return: True if Dockerfile contains RUN dnf instructions in one row
+                False if Dockerfile contains RUN dnf instructions in more rows
+        """
+        value = 0
+        for struct in self.dfp_structure:
+            if struct.get(INSTRUCT) == RUN and "dnf" in struct.get("value"):
+                value += 1
+        if int(value) > 1:
+            return False
+        return True
+
+    def check_chained_run_rest_commands(self):
```

Comment:
I think this test may produce some false positives: sometimes it's convenient to have multiple RUN instructions (which operate on files added to the container)